### PR TITLE
Use the isoband package to calculate contour lines and bands

### PR DESCRIPTION
**Pull request by paleolimbot**
This is a PR to address #3044. While the desired behaviour (calculating contours at draw time that can be labeled) can't be done immediately because `Geom`s don't have access to non-position scales, a good start is to use @clauswilke's [isoband package](http://github.com/clauswilke/isoband) to calculate contour lines (as opposed to `grDevices::contourLines()`) in `stat_contour()`. The isoband package also has a function to return contour polygons, which are implemented in this PR as `stat_contour_filled()` and `geom_contour_filled()`. 

Additional refactoring steps were:

- The `complete` parameter is not used in the current calculation and hasn't been used for about 4 years (estimated from the git blame view), so I removed it.
- I added function parameters and documentation for `bins`, `binwidth` and `breaks`. These appeared in examples but were not function arguments and were not documented.

Things I am not sure about:

- The calculation of breaks in this way is different than the calculation of bins and breaks everywhere else in ggplot2, and I wonder if they should be updated in this PR. In particular, letting breaks be a function I think would be helpful.
- I'm not sure about how I implemented `geom_contour_filled()`. There is no `GeomContourFilled`, it just calls `layer()` with `stat = "contour_filled"` and `geom = "polygon"`.
- I'm not sure what the corner cases are for `isoband::iso(band|contour)`, but I think I should probably add more tests to make sure inappropriate/weird inputs are handled properly (such as for #2702).

### CODE REVIEW

**Comment by clauswilke** [comment]
I had a quick look and didn't see anything that concerns me. I'll be traveling extensively over the next two weeks, starting today. Therefore, you probably shouldn't wait for a more careful review from me, unless there's anything you feel uncertain about.

File: DESCRIPTION [hunk 1]
```diff
@@ -55,7 +55,8 @@ Suggests:
     sf (>= 0.7-3),
     svglite (>= 1.2.0.9001),
     testthat (>= 2.1.0),
-    vdiffr (>= 0.3.0)
+    vdiffr (>= 0.3.0),
+    isoband
```
Comment (hadley) [inline DESCRIPTION]: Should be in alphabetical order
